### PR TITLE
Implement vendor ID and device ID properties

### PR DESCRIPTION
**Pull request by luebbers**
Allows enumeration and property retrieval for PCIe vendor and
device IDs.

### CODE REVIEW

File: ase/api/src/enum.c [hunk 1]
```diff
@@ -781,18 +781,34 @@ fpgaPropertiesSetBBSVersion(fpga_properties prop, fpga_version bbs_version)
...
-	UNUSED_PARAM(vendor_id);
-	FPGA_MSG("Vendor ID not supported");
-	return FPGA_NOT_SUPPORTED;
+	struct _fpga_properties *_prop = (struct _fpga_properties *) prop;
+	fpga_result result = FPGA_INVALID_PARAM;
+	if (NULL == _prop || NULL == vendor_id) {
+		FPGA_ERR("Attempting to dereference NULL pointer(s)");
+		return result;
+	}
+	if (FIELD_VALID(_prop, FPGA_PROPERTY_VENDORID)) {
```
Comment (zhangzhang10) [inline ase/api/src/enum.c]: Do you also need to check if (_prop->magic != FPGA_PROPERTY_MAGIC) ? I see this check in fpgaPropertiesGetDeviceID().

File: libopae/src/enum.c [hunk 2]
```diff
@@ -547,6 +567,22 @@ enum_top_dev(const char *sysfspath, struct dev_list *list)
...
 	pdev->bus = (uint8_t) b;
 
+	// read the vendor and device ID from the 'device' path
+	uint32_t x = 0;
+	char vendorpath[SYSFS_PATH_MAX];
+	snprintf_s_s(vendorpath, SYSFS_PATH_MAX, "%s/device/vendor", sysfspath);
+	result = sysfs_read_u32(vendorpath, &x);
+	if (result != FPGA_OK)
+		return result;
+	pdev->vendor_id = (uint16_t)x;
```
Comment (zhangzhang10) [inline libopae/src/enum.c]: Casting to a narrower int type may upset the compiler, and probably klockwork?
Same comment applies to other similar casts in this PR.

File: libopae/src/properties.c [hunk 3]
```diff
@@ -341,6 +341,24 @@ fpgaUpdateProperties(fpga_token token, fpga_properties prop)
...
 		SET_FIELD_VALID(&_iprop, FPGA_PROPERTY_OBJECTID);
 
+	// read the vendor and device ID from the 'device' path
+	uint32_t x = 0;
+	char vendorpath[SYSFS_PATH_MAX];
+	snprintf_s_s(vendorpath, SYSFS_PATH_MAX, "%s/../device/vendor", _token->sysfspath);
+	result = sysfs_read_u32(vendorpath, &x);
+	if (result != FPGA_OK)
+		return result;
+	_iprop.vendor_id = (uint16_t)x;
```
Comment (zhangzhang10) [inline libopae/src/properties.c]: Casting to a narrower int type may upset the compiler and probably klockwork?
Same comment applies to other similar casts in this PR.

Comment (luebbers) [reply libopae/src/properties.c]: It's an explicit cast and should therefore not upset the analysis tools.

**Pull request merged**